Mac-App/Backend/Backend/BackendAPI/Middleware/ParseSwiftDepsSideEffects/ParseSwiftDepsSideEffectsSpec.swift

```diff
@@ -0,0 +1,141 @@
+                        let sideEffect = sutMiddleware(dispatchFuntion) { AppState.initialState }(nextActionFunction)
+                        sideEffect(SwiftDepsAction.parseFrom(paths: []))
+
+                        expect(actionRecorder.isEmpty).toEventually(beFalse(), timeout: 0.2)
+
+                        expect(actionRecorder.count == expectedNumberOfActions).toEventually(
+                            beTrue(), timeout: 0.2, description: "Expected 1 action to have been dispatched!"
+                        )
+
+                        expect(nextActionRecorder.isEmpty).toEventually(
+                            beFalse(), timeout: 0.2, description: "Side effects must never swallow actions!"
+                        )
+                    }
+                }
+
+                context("when NO url got parsed") {
+                    it("triggers 1 action") {
+                        let expectedNumberOfActions = 1
+
+                        stubbedSwiftDeps = []
+                        let sutMiddleware = sut(mockYamlParser(from:))
+                        let sideEffect = sutMiddleware(dispatchFuntion) { AppState.initialState }(nextActionFunction)
+                        sideEffect(SwiftDepsAction.parseFrom(paths: []))
+
+                        expect(actionRecorder.isEmpty).toEventually(beFalse(), timeout: 0.2)
+
+                        expect(actionRecorder.count == expectedNumberOfActions).toEventually(
+                            beTrue(), timeout: 0.2, description: "Expected 1 action to have been dispatched!"
+                        )
+
+                        expect(nextActionRecorder.isEmpty).toEventually(
+                            beFalse(), timeout: 0.2, description: "Side effects must never swallow actions!"
+                        )
+                    }
+                }
+            }
+
+            context("when SwiftDepsAction.set action gets dispatched") {
+                it("triggers 1 action") {
+                    let expectedNumberOfActions = 1
+
+                    let sutMiddleware = sut(mockYamlParser(from:))
+                    let sideEffect = sutMiddleware(dispatchFuntion) { AppState.initialState }(nextActionFunction)
+                    sideEffect(SwiftDepsAction.set([SwiftDeps.AppDelegateDeps_Fixture]))
+
+                    expect(actionRecorder.isEmpty).toEventually(beFalse(), timeout: 0.2)
+
+                    expect(actionRecorder.count == expectedNumberOfActions).toEventually(
+                        beTrue(), timeout: 0.2, description: "Expected 1 action to have been dispatched!"
+                    )
+
+                    expect(nextActionRecorder.isEmpty).toEventually(
+                        beFalse(), timeout: 0.2, description: "Side effects must never swallow actions!"
+                    )
+                }
+            }
+
+            context("when SwiftDepsAction.reset gets dispatched") {
+                context("and the action is not handled") {
```

Comment:
I'm not so familiar with specs and their contexts yet but, wouldn't it be fine if these two contexts were just merged as "when SwiftDepsAction.reset gets dispatched and the action is not handled"?

Reply:
Yes. In this case this actually makes no difference. I should probably merge it into a single context. 

Basically you use context to separate out different expectations so it reads nicely when you see the tests output. 
For example: 
`parseSwiftDepsSideEffects` -> `when SwiftDepsAction.reset gets dispatched` -> `and the action is not handled` -> `it does not trigger any new action`

If we were to have a second context, it would read as: 

`parseSwiftDepsSideEffects` -> `when SwiftDepsAction.reset gets dispatched` -> `and the action IS handled` -> `it triggers a new action`

Like I said, for this case this separation does not make sense, so I'll merge them into 1.

Reply:
https://github.com/Thurman1776-/Uncomplect-us/pull/9/commits/667c97639a6d3c6bd528916819b7ee4e06ab05dc